src/node.h

```diff
@@ -548,13 +552,14 @@ NODE_EXTERN NODE_DEPRECATED("Use AsyncHooksGetTriggerAsyncId(isolate)",
  * The `trigger_async_id` parameter should correspond to the resource which is
  * creating the new resource, which will usually be the return value of
  * `AsyncHooksGetTriggerAsyncId()`. */
-NODE_EXTERN async_uid EmitAsyncInit(v8::Isolate* isolate,
-                                    v8::Local<v8::Object> resource,
-                                    const char* name,
-                                    async_uid trigger_async_id);
+NODE_EXTERN async_context EmitAsyncInit(v8::Isolate* isolate,
+                                        v8::Local<v8::Object> resource,
+                                        const char* name,
+                                        async_id trigger_async_id = -1);
```

Comment:
I don’t really understand the question, sorry. Basically, what we need to shim is the old definition
```c++
async_uid EmitAsyncInit(v8::Isolate* isolate,
                        v8::Local<v8::Object> resource,
                        const char* name,
                        async_uid trigger_async_id);
```
in terms of the new one in this PR, right?

So what’s speaking against `OldStyleEmitAsyncInit(resource, name, trigger_async_id) := (NewStyleAsyncInit(resource, name, trigger_async_id)).async_id_`?

Or are you thinking about the return type? The hacky solution would be to enable casting from `async_context` to `async_id` by returning `async_id_` (and immediately deprecating that), it’s not pretty but as long as it doesn’t end up in master I’m okay with that.

(One more thing: I’m noticing, maybe we should not underscore-suffix the fields of `async_context` given that they are public, not internal…)